Tests/Common/Securities/OptionStrategyPositionGroupBuyingPowerModelTests.cs

```diff
@@ -54,6 +58,259 @@ public void Setup()
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 0, 1000, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 0, 10000, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 0, -310, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 0, -330, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 20, 1000 - 20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 20, 10000 - 20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 20, -20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 20, -310 - 20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, 20, -330 - 20, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, -310 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, -330 - -20, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, 20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, 1000 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, 10000 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 0, 980, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 0, 1000, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 0, -980, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 0, -1000, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, 980 - 20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, 1000 - 20, false),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, -20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, -980 - 20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, -1000 - 20, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, -980 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, -1000 - -20, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, 20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, 980 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, 1000 - -20, false).Explicit(),
+        };
+
+        [TestCaseSource(nameof(HasSufficientBuyingPowerForOrderTestCases))]
+        public void HasSufficientBuyingPowerForOrder(OptionStrategyDefinition optionStrategy, int initialPositionQuantity, int orderQuantity,
+            bool expectedResult)
+        {
+            _algorithm.SetCash(1000000);
```

Comment:
`_algorithm.SetCash(1000000);` this is already set I think?

Reply:
Just making sure that accidentally changing it on the SetUp method won't affect this test.